src/ipc/dbus/ipc-dbus.h

```diff
@@ -978,6 +977,73 @@ class DBusIPCAttachedPropertyFactory : public IPCAttachedPropertyFactoryBase
 };
 
 }
+
+
+class InterfacePropertyHandlerBase {
+public:
+
+    QString generateObjectPath(const QString& parentPath) {
+        QString path = parentPath + "/dynamic";
+        path += QString::number(s_nextInstanceID++);
+        return path;
+    }
+
+    static int s_nextInstanceID;
+};
+
+
+template<typename InterfaceType, typename InterfaceAdapterType>
+class InterfacePropertyIPCAdapterHandler : public InterfacePropertyHandlerBase {
+
+public:
+    void update(IPCServiceAdapterBase* parent, InterfaceType* service) {
+        if (m_service != service) {
+            m_service = service;
+            auto serviceAdapter = new InterfaceAdapterType(service);
+            serviceAdapter->setObjectPath(generateObjectPath(parent->objectPath()));
+            serviceAdapter->setService(service);
+            serviceAdapter->init();
+            m_serviceAdapter = std::move(std::unique_ptr<InterfaceAdapterType>(serviceAdapter));
+        }
+    }
+
+    QString objectPath() const {
+        if (m_serviceAdapter) {
```

Comment:
also format, but better use tertiary operator, wouldn't QString() be better thatn ""?